bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/TreeViewer.java

```diff
@@ -1203,4 +1213,39 @@ Object getLastElement(Widget parent) {
 		}
 		return items[length - 1].getData();
 	}
+
+	@Override
+	void handleKeySelection(KeyEvent e) {
+		if (e.keyCode == SWT.KEYPAD_MULTIPLY || e.keyCode == SWT.KEYPAD_ADD || e.keyCode == SWT.ARROW_RIGHT) {
+			if (e.widget instanceof Tree table) {
+				TreeItem[] selection = tree.getSelection();
+				if (selection.length != 1) {
+					return;
+				}
+				handleExpandableNodeClicked(selection[0], false);
+			}
+		}
+	}
+
+	@Override
+	void handleMenuSelection(MenuEvent e, Menu popupMenu) {
+		if (tree.getSelection().length != 1) {
+			return;
+		}
+		TreeItem selectedItem = tree.getSelection()[0];
+		if (selectedItem.getData() instanceof ExpandableNode) {
+			MenuItem[] items = popupMenu.getItems();
+			for (int i = 0; i < items.length; i++) {
+				items[i].dispose();
+			}
+			MenuItem newItem = new MenuItem(popupMenu, SWT.NONE);
+			newItem.setText("Expand All Remaining Items"); //$NON-NLS-1$
```

Comment:
Two things:
1) The text should be externalized.
2) "Expand All Remaining Items" ->  "Show All Remaining Items"